scale/queue/test/test_models.py

```diff
@@ -369,114 +216,189 @@ def setUp(self):
+                        'job_type_name': self.job_type_2.name,
+                        'job_type_version': self.job_type_2.version,
+                        'job_type_revision': self.job_type_2.revision_num
+                    }
+                }
+            }
         }
 
         recipe_definition = RecipeDefinition(definition)
-        recipe_definition.validate_job_interfaces()
 
         self.recipe_type = recipe_test_utils.create_recipe_type_v6(definition=definition)
 
-        data = {
-            'version': '1.0',
-            'input_data': [{
-                'name': 'Recipe Input',
-                'file_id': source_file.id,
-            }],
-            'workspace_id': workspace.id,
-        }
-        self.data = LegacyRecipeData(data)
-
-    @patch('queue.models.CommandMessageManager')
-    def test_successful(self, mock_msg_mgr):
         workspace = storage_test_utils.create_workspace()
-        source_file = source_test_utils.create_source(workspace=workspace)
-        event = trigger_test_utils.create_trigger_event()
-        recipetype1 = recipe_test_utils.create_recipe_type_v6()
+        strike_source_file = source_test_utils.create_source(workspace=workspace)
+        scan_source_file = source_test_utils.create_source(workspace=workspace)
 
         data_dict = {
             'version': '1.0',
             'input_data': [{
-                'name': 'input_a',
-                'file_id': source_file.id,
+                'name': 'INPUT_IMAGE',
+                'file_id': strike_source_file.id,
             }],
             'output_data': [{
                 'name': 'output_a',
                 'workspace_id': workspace.id
             }]
         }
-        data = JobDataV6(data_dict)
+        self.data = JobDataV6(data_dict)
+
+    @patch('queue.models.CommandMessageManager')
+    def test_successful(self, mock_msg_mgr):
+        workspace = storage_test_utils.create_workspace()
+        strike_source_file = source_test_utils.create_source(workspace=workspace)
+        scan_source_file = source_test_utils.create_source(workspace=workspace)
+        event = trigger_test_utils.create_trigger_event()
+        recipetype1 = recipe_test_utils.create_recipe_type_v6()
+
+        data = Data()
+        data.add_value(FileValue('input_a', [123]))
 
         config_dict = {'version': '6',
                        'output_workspaces': {'default': workspace.name},
                        'priority': 999}
         config = RecipeConfigurationV6(config_dict).get_configuration()
 
-        created_recipe = Queue.objects.queue_new_recipe_v6(recipetype1, data._new_data, event, recipe_config=config)
+        created_recipe = Queue.objects.queue_new_recipe_v6(recipetype1, data, event, recipe_config=config)
 
-        self.assertDictEqual(created_recipe.configuration, config_dict)
+        data = Data()
+        data.add_value(FileValue('input_a', [123]))
+        created_recipe = Queue.objects.queue_new_recipe_v6(recipetype1, data, event)
```

Comment:
Looks like a merge issue?  Shouldn't we be testing the created recipe rather than simply creating it again without a config?

Reply:
Yes, merge issue. Good catch, thank you.